src/puppetlabs/puppetdb/query/events.clj

```diff
@@ -50,49 +50,74 @@
 (defn with-latest-events
   "CTE to wrap unioned queries when distinct_resources is used"
   [query]
-  (format
-    "WITH latest_events AS
-           (SELECT certnames.certname,
-                   resource_events.resource_type,
-                   resource_events.resource_title,
-                   resource_events.property,
-                   MAX(resource_events.timestamp) AS timestamp
-                   FROM resource_events
-                   JOIN certnames ON resource_events.certname_id = certnames.id
-                   WHERE resource_events.timestamp >= ?
-                   AND resource_events.timestamp <= ?
-                   GROUP BY certname, resource_type, resource_title, property)
-           %s" query))
+  (let [collate (if (sutils/postgres?) "COLLATE \"C\"" "")]
+    (format
+      "WITH latest_events AS
+       (SELECT certname,
```

Comment:
left some discussion of this in PE-11006. Short story is that I like the direction, but the perf benefits don't carry over to queries on shorter time intervals. I think we should explore this as a separate ticket, assuming that the current work gets us in passable shape. @kbrezina sound right to you?